tests/Agent/IntegrationTests/IntegrationTests/Errors/ExpectedErrorTests.cs

```diff
@@ -77,16 +77,17 @@ public void Test()
             };
 
             var errorTraces = _fixture.AgentLog.GetErrorTraces().ToList();
-            var errorEvents = _fixture.AgentLog.GetErrorEvents().ToList();
+            var errorEventPayloads = _fixture.AgentLog.GetErrorEvents().ToList();
+            var errorEvents = errorEventPayloads.SelectMany(payload => payload.Events).ToList();
```

Comment:
Is there a way to build this into the Agent Log feature for all event types?   Seems like this functionality could be useful to many tests in making them less brittle.

Reply:
It's a fair point.  I did consider doing something like this.  A lot of tests use the `GetErrorEvents()` method so it would require changing a lot of test code.

One thing approach I'd explore if we decided to spend more time on this, is rename `GetErrorEvents` to `GetErrorEventPayloads`, because that's what it's ACTUALLY returning.  Then I'd probably need to go look at each test which uses that method and see what it's actually trying to assert on (for example, if it's just asserting on the absence of error events, then making sure that there are no error event payloads is correct, but if it really wants to know about error events than it might make sense to use a new method which returns the actual events).